lib/tasks/bower-install.js

```diff
@@ -2,56 +2,28 @@
 
 // Runs `bower install` in cwd
 
-var Promise = require('../ext/promise');
-var Task    = require('../models/task');
+var BowerTask   = require('./bower-task');
+var _           = require('lodash');
 
-module.exports = Task.extend({
-  init: function() {
-    this.bower = this.bower || require('bower');
-    this.bowerConfig = this.bowerConfig || require('bower-config');
-  },
-  // Options: Boolean verbose
-  run: function(options) {
-    var chalk          = require('chalk');
-    var bower          = this.bower;
-    var bowerConfig    = this.bowerConfig;
-    var ui             = this.ui;
-    var packages       = options.packages || [];
-    var installOptions = options.installOptions || { save: true };
+module.exports = BowerTask.extend({
+  command: 'install',
+  startProgressMessage: 'Installing browser packages via Bower',
+  completionMessage: 'Installed browser packages via Bower.',
 
-    ui.startProgress(chalk.green('Installing browser packages via Bower'), chalk.green('.'));
+  buildOptions: function(options) {
+    var superOptions = this._super.buildOptions.call(this, options);
 
-    var config = bowerConfig.read();
-    config.interactive = true;
+    var installOptions = options.installOptions || {};
+    if (options.packages && typeof installOptions.save === 'undefined') {
+      installOptions.save = true;
+    }
+    return _.extend(superOptions, installOptions);
+  },
 
-    return new Promise(function(resolve, reject) {
-        bower.commands.install(packages, installOptions, config) // Packages, options, config
-          .on('log', logBowerMessage)
-          .on('prompt', ui.prompt.bind(ui))
-          .on('error', reject)
-          .on('end', resolve);
-      })
-      .finally(function() { ui.stopProgress(); })
-      .then(function() {
-        ui.writeLine(chalk.green('Installed browser packages via Bower.'));
-      });
+  buildArgs: function(options) {
+    var superArgs = this._super.buildArgs.call(this.options) || [];
+    var packages = options.packages || [];
 
-    function logBowerMessage(message) {
-      if (message.level === 'conflict') {
-        // e.g.
-        //   conflict Unable to find suitable version for ember-data
-        //     1) ember-data 1.0.0-beta.6
-        //     2) ember-data ~1.0.0-beta.7
-        ui.writeLine('  ' + chalk.red('conflict') + ' ' + message.message);
-        message.data.picks.forEach(function(pick, index) {
-          ui.writeLine('    ' + chalk.green((index + 1) + ')') + ' ' +
-                       message.data.name + ' ' + pick.endpoint.target);
-        });
-      } else if (message.level === 'info' && options.verbose) {
-        // e.g.
-        //   cached git://example.com/some-package.git#1.0.0
-        ui.writeLine('  ' + chalk.green(message.id) + ' ' + message.message);
-      }
-    }
+    return _.compact([].concat(superArgs, packages));
```

Comment:
`buildOptions`/`buildArgs` implementations looks difficult to me. @stefanpenner @kellyselden do you have any ideas on how to make it more simple or reasonable?